src/cls/CMakeLists.txt

```diff
@@ -1,152 +1,220 @@
 
-add_library(cls_user_client STATIC
+set(cls_user_client_srcs
   user/cls_user_client.cc
   user/cls_user_types.cc
   user/cls_user_ops.cc)
+add_library(cls_user_client STATIC ${cls_user_client_srcs})
+
+list(APPEND cls_embedded_srcs ${cls_user_srcs} ${cls_user_client_srcs})
 
 # cls_journal
-add_library(cls_journal SHARED
+set(cls_journal_srcs
   journal/cls_journal.cc
   journal/cls_journal_types.cc)
+add_library(cls_journal SHARED ${cls_journal_srcs})
 set_target_properties(cls_journal PROPERTIES VERSION "1.0.0" SOVERSION "1")
 install(TARGETS cls_journal DESTINATION ${cls_dir})
 
-add_library(cls_journal_client STATIC
+set(cls_journal_client_srcs
   journal/cls_journal_client.cc
   journal/cls_journal_types.cc)
+add_library(cls_journal_client STATIC ${cls_journal_client_srcs})
+
+list(APPEND cls_embedded_srcs ${cls_journal_srcs} ${cls_journal_client_srcs})
 
 # cls_rgw
 if (WITH_RADOSGW)
-  add_library(cls_rgw SHARED
+  set(cls_rgw_srcs
     rgw/cls_rgw.cc
     rgw/cls_rgw_ops.cc
     rgw/cls_rgw_types.cc
     ${CMAKE_SOURCE_DIR}/src/common/ceph_json.cc)
+  add_library(cls_rgw SHARED ${cls_rgw_srcs})
   target_link_libraries(cls_rgw json_spirit)
   set_target_properties(cls_rgw PROPERTIES VERSION "1.0.0" SOVERSION "1")
   install(TARGETS cls_rgw DESTINATION ${cls_dir})
 
-  add_library(cls_rgw_client STATIC
+  set(cls_rgw_client_srcs
     rgw/cls_rgw_client.cc
     rgw/cls_rgw_types.cc
     rgw/cls_rgw_ops.cc)
+  add_library(cls_rgw_client STATIC ${cls_rgw_client_srcs})
+
+  list(APPEND cls_embedded_srcs ${cls_rgw_srcs} ${cls_rgw_client_srcs})
 endif (WITH_RADOSGW)
 
 # cls_cephfs
 if (WITH_CEPHFS)
-  add_library(cls_cephfs SHARED
+  set(cls_cephfs_srcs
     cephfs/cls_cephfs.cc)
+  add_library(cls_cephfs SHARED ${cls_cephfs_srcs})
   set_target_properties(cls_cephfs PROPERTIES VERSION "1.0.0" SOVERSION "1")
   install(TARGETS cls_cephfs DESTINATION ${cls_dir})
 
-  add_library(cls_cephfs_client STATIC
+  set(cls_cephfs_client_srcs
     cephfs/cls_cephfs_client.cc)
-endif (WITH_CEPHFS)
+  add_library(cls_cephfs_client STATIC ${cls_cephfs_client_srcs})
 
-add_library(cls_numops_client STATIC numops/cls_numops_client.cc)
+  list(APPEND cls_embedded_srcs ${cls_cephfs_srcs} ${cls_cephfs_client_srcs})
+endif (WITH_CEPHFS)
 
-add_library(cls_lua SHARED
+# cls_lua
+set(cls_lua_srcs
     lua/cls_lua.cc
-    lua/lua_bufferlist.cc
-)
+    lua/lua_bufferlist.cc)
+add_library(cls_lua SHARED ${cls_lua_srcs})
 set_target_properties(cls_lua PROPERTIES VERSION "1.0.0" SOVERSION "1")
 install(TARGETS cls_lua DESTINATION ${cls_dir})
 target_link_libraries(cls_lua
     liblua
-    json_spirit
-)
-
-add_library(cls_lua_client STATIC
-    lua/cls_lua_client.cc
-)
+    json_spirit)
+
+set(cls_lua_client_srcs
+    lua/cls_lua_client.cc)
+add_library(cls_lua_client STATIC ${cls_lua_client_srcs})
+
+list(APPEND cls_embedded_srcs ${cls_lua_srcs} ${cls_lua_client_srcs})
+
+if(WITH_EMBEDDED)
+  include(MergeStaticLibraries)
+  list(REMOVE_DUPLICATES cls_embedded_srcs)
```

Comment:
Is this REMOVE_DUPLICATES a backup in case something goes wrong? I'm not seeing why this is necessary.